File: SortingAlgorithms/BubbleSort.java
package SortingAlgorithms;

public class BubbleSort {

    BubbleSort() {}

    private int[] sort(int[] arr) {

		boolean swapped = false; 
		int n = arr.length - 1; 
        for (int i = 0; i < n; i++) {
			
			swapped = false;

			for (int j = 0; j < n - i; j++) {
				
	            if (arr[j] > arr[j+1]) {
					int temp = arr[j];
	            	arr[j] = arr[j+1];
	            	arr[j+1] = temp;
	            	swapped = true;
	            }
			}
			
			if (!swapped) {
				break; 
			}            
        }
     	return arr;
    }

    public static void main(String[] args) {
		int[] arr = {2, 4, 1, 10, 2};

		BubbleSort bubbleSort = new BubbleSort();
		for (int ele: arr) {
			System.out.print(ele + " ");
		}

		System.out.println();
		int[] newArr = bubbleSort.sort(arr);
		
		for (int ele: newArr) {
			System.out.print(ele + " ");
		}
		System.out.println();
			
    }
}

File: SortingAlgorithmsImpl/BubbleSort.cpp
#include <iostream>
#include <vector>

using namespace std;

class BubbleSort {

	public:

		void sort(vector<int> &vec) {

			for(size_t i = 0; i < vec.size(); i++) {
				bool swapped = false; 

				for (size_t j= 0; j < vec.size() - i - 1; j++) {

					if (vec[j] > vec[j+1]) {
						int temp = vec[j+1];
						vec[j+1] = vec[j];
						vec[j] = temp;
						swapped = true;
					}
				}

				if (!swapped) {
					break;
				}
			}
		}
	
};

int main() {

	BubbleSort bubbleSort;

	vector<int> vec = {32, 1, 12, 24, 63, 5, 3, 1};

	for (size_t i = 0; i < vec.size(); i++) {
		cout << vec[i] << " ";
	}

	cout << endl;
	
	bubbleSort.sort(vec);

	cout << "\n Bubble sorting" << endl;
	
	for (size_t i = 0; i < vec.size(); i++) {
		cout << vec[i] << " ";
	}
	
}
